ecosystem/purescript-0.14-migration-guide.md

```diff
@@ -0,0 +1,618 @@
+
+#### The dependency graph for `functors` and related packages have changed in order to unify newtypes among the libraries.
+
+Implemented in [purescript-foldable-traversable#131](https://github.com/purescript/purescript-foldable-traversable/pull/131), [purescript-functors#31](https://github.com/purescript/purescript-functors/pull/31), [purescript-profunctor#41](https://github.com/purescript/purescript-profunctor/pull/41), [purescript-distributive#17](https://github.com/purescript/purescript-distributive/pull/17), [purescript-contravariant#30](https://github.com/purescript/purescript-contravariant/pull/30), [purescript-bifunctors#22](https://github.com/purescript/purescript-bifunctors/pull/22), [purescript-tuples#46](https://github.com/purescript/purescript-tuples/pull/46), [purecsript-identity#26](https://github.com/purescript/purescript-identity/pull/26), [purescript-either#64](https://github.com/purescript/purescript-either/pull/64), and [purecsript-const#19](https://github.com/purescript/purescript-const/pull/19). 
+
+Mostly discussed in [purescript-profunctor#23](https://github.com/purescript/purescript-profunctor/issues/23).
+
+A number of dependencies have been rearranged in the core libraries so that more types can be reused among them (especially newtypes between `profunctor` and `bifunctors`) and so that fundamental data types like `either` and `tuple` are at the bottom of the dependency hierarchy.
+
+Most of these changes involve instances and should be invisible to user code. However, some functions and modules have changed locations and will require changes in your code if you use them.
+
+**To fix:**
+
+The `lookup` function has moved from `Data.Tuple` to `Data.Foldable` and the import should be updated accordingly:
+
+```purs
+- import Data.Tuple (lookup)
++ import Data.Foldable (lookup)
+```
+
+Several modules have moved or been renamed, all from the `profunctors` and `bifunctors` packages into the `functors` package. You'll need to update your imports accordingly:
+
+```
+Data.Profunctor.Clown  -> Data.Functor.Clown
+Data.Profunctor.Costar -> Data.Functor.Costar
+Data.Profunctor.Cowrap -> Data.Functor.Flip
+Data.Profunctor.Joker  -> Data.Functor.Joker
+Data.Bifunctor.Product -> Data.Functor.Product2
+```
+
+The `Data.Profunctor.Wrap` and `Data.Bifunctor.Wrap` types have been deleted, as any profunctor or bifunctor should also have a functor instance.
+
+This was the old dependency graph:
+
+![old-dependency-graph](https://user-images.githubusercontent.com/1570964/106351480-6e55a100-62aa-11eb-8e8a-72cb33e354dc.png)
+
+This is the new dependency graph, where the dotted line represents three packages which may be merged together in the future:
+
+![new-dependency-graph](https://user-images.githubusercontent.com/1570964/106351479-6ac21a00-62aa-11eb-8234-803f7868cc53.png)
```

Comment:
I think these graphs should be shown first in this section.

Reply:
I actually don't think we should show people these dependency graphs; the vast majority of users shouldn't need to care about them, I think.

Reply:
Upon more reflection, I think the graph summarizes some changes (e.g. instances were moved) far quicker than words do. However, it doesn't summarize the heart of the change faster than words do, which is that we don't have two newtypes anymore (e.g. `Joker`, etc).

So, I yield. Harry's point is better.